internal/datastore/common/sql.go

```diff
@@ -325,12 +351,18 @@ func (sqf SchemaQueryFilterer) MustFilterWithRelationshipsFilter(filter datastor
 }
 
 func (sqf SchemaQueryFilterer) FilterWithRelationshipsFilter(filter datastore.RelationshipsFilter) (SchemaQueryFilterer, error) {
-	sqf = sqf.FilterToResourceType(filter.ResourceType)
+	if filter.OptionalResourceType != "" {
+		sqf = sqf.FilterToResourceType(filter.OptionalResourceType)
```

Comment:
`sqf` is the method received and is not a pointer. I assume this is not intended to have side effects and is just used for ergonomic convenience. To make it more obvious, create a local variable that does not shadow the receiver.